infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -86,10 +104,12 @@ public Set<String> getStreamsToReplicate() {
             openExistingTable(LOG_REPLICATION_PLUGIN_VERSION_TABLE);
             openExistingTable(LOG_REPLICATION_STREAMS_NAME_TABLE);
             if (!tableVersionMatchesPlugin()) {
-                // delete the tables and recreate them
+                // Case: Upgrade - delete the tables and recreate them
                 deleteExistingStreamNameAndVersionTables();
                 createStreamNameAndVersionTables(
                     logReplicationConfigAdapter.fetchStreamsToReplicate());
+                // Persist upgrade flag so a snapshot-sync is enforced upon negotiation
+                updateUpgradeFlag(true);
```

Comment:
If LR crashes after delete and create of the tables is complete but before the upgrade flag is set, we will not force the snapshot sync subsequently.  So all these 3 operations must be done atomically in the same transaction.

Reply:
Have moved this to the table creation transaction, thanks!